transcom/mymove
# INT B-18831 admin UI add approve reject buttons and rejection input on overview page of requested office user (MERGE AFTER B-18829)

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18831)

## Summary

**NOTE** this does contain the branch of INT B-18829 https://github.com/transcom/mymove/pull/12294 in order to get the full flow of things.

When the admin user needs to view the overview of a requested office user, we want them to have some quick & dirty ways to approve/reject users that do not need any changes. 

This PR does the following for the single requested office user overview page:
- adds approve button to overview page
- adds reject button to overview page
- adds rejection reason input to overview page
- adds edit button to overview page
- added `adminApi.js` to allow for calling of the admin API from the UI
- changed swagger information to be more updated and use common format of GHC swagger docs
- added CSS where appropriate
- added `email` to payload to support future work in [B-18828](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18828)

### How to test

1. Access MM as an admin user
2. Navigate to the requested office user list
3. If you do not see any, you'll need to update the status to `REQUESTED` of some rows in the `office_users` table
4. Click on a user
5. You should see the overview page with the added functionality of a rejection reason input and approve/reject buttons
6. When clicking approve, this updates the user to APPROVED status, vice versa for REJECTED button
7. You can provide a rejection reason if needed
8. Following testing, please ensure values update in the database

## Screenshots
![Screenshot 2024-03-21 at 3 28 35 PM](https://github.com/transcom/mymove/assets/136510600/26a55140-ec42-48ac-978e-67e4f802ce4b)

![Screenshot 2024-03-21 at 3 23 14 PM](https://github.com/transcom/mymove/assets/136510600/f6759afd-8e4a-4d94-b1e8-c8d795f597bf)
